### PR TITLE
Add example of goldenFileComparator usage in widget tests

### PR DESCRIPTION
**Pull request by alestiago**
As from #76337, precisely https://github.com/flutter/flutter/issues/76337#issuecomment-795486138 and https://github.com/flutter/flutter/issues/76337#issuecomment-2172941109. There is missing documentation with an example on how the [`goldenFileComparator`](https://api.flutter.dev/flutter/flutter_test/goldenFileComparator.html) can be used.

This change updates the documentation by providing an additional code snippet to the example that was already stated by the documentation. 

Follows the effort made by #76337 and https://github.com/flutter/flutter/pull/150343.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_test/lib/src/goldens.dart [hunk 1]
```diff
@@ -120,8 +120,61 @@ abstract class GoldenFileComparator {
...
+///     find.byType(ColoredBox),
+///     matchesGoldenFile('my_golden.png'),
+///   );
+/// });
+///
+/// class _TolerantGoldenFileComparator extends LocalFileComparator {
+///   _TolerantGoldenFileComparator(
+///     super.testFile, {
+///     required double precisionTolerance,
+///   }) : _precisionTolerance = precisionTolerance;
```
Comment (Piinks) [inline packages/flutter_test/lib/src/goldens.dart:150]:
```suggestion
///   }) : assert(precisionTolerance <= 1 || precisionTolerance  0),
            _precisionTolerance = precisionTolerance;
```

Based on the dos below. :)

**Review by Piinks — COMMENTED** [review]
Hey @alestiago thanks for adding an example of this.

**Review by Piinks — APPROVED** [review]
LGTM, thank you!

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks for the contribution:)

To fix the linux analyzer, ~we might need to add some comments at top of the file to add some assumption for the example.~ Please see the suggested changes below. I think these changes can help fix the linux analyze test:)

File: packages/flutter_test/lib/src/goldens.dart [hunk 2]
```diff
@@ -120,8 +120,65 @@ abstract class GoldenFileComparator {
...
+///
+///   /// How much the golden image can differ from the test image.
+///   ///
+///   /// It is expected to be between 0 and 1. Where 0 is no difference (the same image)
+///   /// and 1 is the maximum difference (completely different images).
+///   final double _precisionTolerance;
+///
+///   @override
+///   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+///     final result = await GoldenFileComparator.compareLists(
```
Comment (QuncCccccc) [inline packages/flutter_test/lib/src/goldens.dart:164]:
```suggestion
///     final ComparisonResult result = await GoldenFileComparator.compareLists(
```

File: packages/flutter_test/lib/src/goldens.dart [hunk 3]
```diff
@@ -120,8 +120,65 @@ abstract class GoldenFileComparator {
...
+///       imageBytes,
+///       await getGoldenBytes(golden),
+///     );
+///
+///     final passed = result.passed || result.diffPercent <= _precisionTolerance;
```
Comment (QuncCccccc) [inline packages/flutter_test/lib/src/goldens.dart:169]:
```suggestion
///     final bool passed = result.passed || result.diffPercent <= _precisionTolerance;
```

File: packages/flutter_test/lib/src/goldens.dart [hunk 4]
```diff
@@ -120,8 +120,65 @@ abstract class GoldenFileComparator {
...
+///     if (passed) {
+///       result.dispose();
+///       return true;
+///     }
+///
+///     final error = await generateFailureOutput(result, golden, basedir);
```
Comment (QuncCccccc) [inline packages/flutter_test/lib/src/goldens.dart:175]:
```suggestion
///     final String error = await generateFailureOutput(result, golden, basedir);
```

File: packages/flutter_test/lib/src/goldens.dart [hunk 5]
```diff
@@ -120,8 +120,65 @@ abstract class GoldenFileComparator {
...
+///   );
+///   addTearDown(() => goldenFileComparator = previousGoldenFileComparator);
+///
+///   await tester.pumpWidget(const ColoredBox(color: Color(0xff00ff00)));
+///
+///   await expectLater(
+///     find.byType(ColoredBox),
+///     matchesGoldenFile('my_golden.png'),
+///   );
+/// });
```
Comment (QuncCccccc) [inline packages/flutter_test/lib/src/goldens.dart:144]: I think use `void main() {}` to wrap the `testWidgets` method, and declare the types of variables(here and below) can fix the problem:
```suggestion
/// void main() {
///   testWidgets('matches golden file with a 0.01 tolerance', (WidgetTester tester) async {
///     final GoldenFileComparator previousGoldenFileComparator = goldenFileComparator;
///     goldenFileComparator = _TolerantGoldenFileComparator(
///       Uri.parse('test/my_widget_test.dart'),
///       precisionTolerance: 0.01,
///     );
///     addTearDown(() => goldenFileComparator = previousGoldenFileComparator);
///
///     await tester.pumpWidget(const ColoredBox(color: Color(0xff00ff00)));
///
///     await expectLater(
///       find.byType(ColoredBox),
///       matchesGoldenFile('my_golden.png'),
///     );
///   });
/// }
```

**Comment by alestiago** [comment]
@Piinks , @QuncCccccc thank you for the reviews! I've comitted the changes suggested by @QuncCccccc in https://github.com/flutter/flutter/pull/150422/commits/95156f90762275de7100c876d4b96fec7d708580 and they have fixed the pipeline. 

I'll let you all merge it in 🙌

**Pull request merged**